HighAvailabilityModule.UnitTest/AlgorithmTest.cs

```diff
@@ -20,46 +21,69 @@ public class AlgorithmTest
         private static DateTime Now { get; } = DateTime.Parse("2019-09-27T12:00:00.2965246Z");
 
         private static string Client1Uuid => "cdca5b45-6ea1-4d91-81f6-d39f4821e791";
+        private static string ClientUtypeA => "A";
+        private static string ClientUnum1 => "1";
 
-        private static string Client2Uuid => "33253ab7-27b6-478c-a359-4eca7df83b80";
+        private static string Client2Uuid => "39a78df0-e101-49b9-8c56-ec2fea2e47df";
+        private static string ClientUnum2 => "2";
+
+        private static string Client3Uuid => "33253ab7-27b6-478c-a359-4eca7df83b80";
+        private static string ClientUtypeB => "B";
 
         private InMemoryMembershipServer server;
 
         private InMemoryMembershipClient client;
         private InMemoryMembershipClient client2;
+        private InMemoryMembershipClient client3;
 
         private MembershipWithWitness algo;
         private MembershipWithWitness algo2;
+        private MembershipWithWitness algo3;
+
+        private void TestFunc(Dictionary<string, HeartBeatEntry> Current, string Uuid, string Utype, string Unum)
+        {
+            Assert.IsTrue(Current != null);
+            Assert.IsTrue(Current.ContainsKey(Utype));
+            Assert.IsTrue(Current[Utype] != null);
+            Assert.IsTrue(Current[Utype].Uuid == Uuid);
+            Assert.IsTrue(Current[Utype].Utype == Utype);
+            Assert.IsTrue(Current[Utype].Unum == Unum);
+        }
 
         [TestInitialize]
         public void Initialize()
         {
             this.server = new InMemoryMembershipServer(Timeout);
-            this.client = new InMemoryMembershipClient(this.server, Client1Uuid);
+            this.client = new InMemoryMembershipClient(this.server, Client1Uuid, ClientUtypeA, ClientUnum1);
             this.algo = new MembershipWithWitness(this.client, Interval, Timeout);
-            this.client2 = new InMemoryMembershipClient(this.server, Client2Uuid);
+
+            this.client2 = new InMemoryMembershipClient(this.server, Client2Uuid, ClientUtypeA, ClientUnum2);
             this.algo2 = new MembershipWithWitness(this.client2, Interval, Timeout);
+
+            this.client3 = new InMemoryMembershipClient(this.server, Client3Uuid, ClientUtypeB, ClientUnum1);
+            this.algo3 = new MembershipWithWitness(this.client3, Interval, Timeout);
         }
 
         [TestMethod]
         public async Task RunningAsPrimaryTest1()
         {
-            this.server.Current = null;
+            this.server.CurrentTable = null;
             Assert.IsFalse(this.algo.RunningAsPrimary(Now));
         }
 
         [TestMethod]
         public async Task RunningAsPrimaryTest2()
         {
-            this.server.Current = null;
+            this.server.CurrentTable = null;
```

Comment:
This case will need to change. The reason is in the old code, `server.Current == null` is a valid state, which is when server just started. But in the new code, `server.Current` will never be null. Can we simply change it to `server.CurrentTable = new Dictionary`?